bokehjs/src/coffee/ar/transforms.coffee

```diff
@@ -0,0 +1,113 @@
+HasProperties = require "../common/has_properties"
+
+class Transform extends HasProperties
+
+class AutoEncode extends Transform
+  type: "AutoEncode"
+
+class BinarySegment extends Transform
+  type: "BinarySegment"
+
+class Const extends Transform
+  type: "Const"
+
+class Contour extends Transform
+  type: "Contour"
+
+class Count extends Transform
+  type: "Count"
+
+class CountCategories extends Transform
+  type: "CountCategories"
+
+class Cuberoot extends Transform
+  type: "Cuberoot"
+
+class Encode extends Transform
+  type: "Encode"
+
+class HDAlpha extends Transform
+  type: "HDAlpha"
+
+class Id extends Transform
+  type: "Id"
+
+class Interpolate extends Transform
+  type: "Interpolate"
+
+class InterpolateColor extends Transform
+  type: "InterpolateColor"
+
+class Log extends Transform
+  type: "Log"
+
+class NonZero extends Transform
+  type: "NonZero"
+
+class Ratio extends Transform
+  type: "Ratio"
+
+class Seq extends Transform
+  type: "Seq"
+
+class Spread extends Transform
+  type: "Spread"
+
+class ToCounts extends Transform
+  type: "ToCounts"
+
+module.exports =
+
+  AutoEncode:
+    Model: AutoEncode
```

Comment:
Yes, it goes back to the cross-language nature. All we have coming is a name, so we map names  to a uniform interface that has an object with `Model` and `Collection` so that we can do:
```
Collections("AutoEncode").Model 
```
as a way to get the `AutoEncode` model, which is distinct from any of the other models we might need to make starting from just a string name.